hoomd/md/pair/aniso.py

```diff
@@ -563,3 +572,425 @@ def type_shapes(self):
+                                     envelope_params = {'alpha': math.pi/4,
+                                                        'omega': 30})
+        patchy.patches.default = []
+        simulation.operations.integrator.forces = [patchy]
+
+    For multiple patches, every combination of patch and
+    We use PatchyLJ as the example.
+
+    To make specific patches
+
+    todo write example
+
+    .. py:attribute:: params
+
+        The Patchy potential parameters. A dictionary of dictionaries that
+        are passed on to the envelope and to the pair potential.
+
+        The dictionary has the following keys:
+
+        * ``envelope_params`` (`dict`, **required**)
+
+          * ``alpha`` (`float`) - patch half-angle :math:`\alpha` :math:`[\mathrm{rad}]`
+          * ``omega`` (`float`) - patch steepness :math:`\omega`
+
+
+        * ``pair_params`` (`dict`, **required**) -
+          passed to isotropic potential (see subclasses).
+
+        .. rubric:: Example:
+
+        .. code-block:: python
+
+            envelope_params = {'alpha': math.pi/4, 'omega': 30}
+            patchy.params[('A', 'A')] = dict(pair_params = pair_params,
+                                             envelope_params = envelope_params)
+
+        Type: `TypeParameter` [`tuple` [``particle_type``, ``particle_type``],
+        `dict`]
+
+    .. py:attribute:: patches
+
+        List of vectors pointing to patch centers. Unnormalized vectors are normalized.
+        If a particle type does not have patches, set the patches to an empty list.
+
+        Type: `list` [`tuple` [`float`, `float`, `float`]] or `list` [()]
+
+        .. rubric:: Example:
+
+        .. code-block:: python
+
+            patchy.patches['A'] = [(1,0,0), (1,1,1)]
+    """
+
+    def __init__(self, nlist, default_r_cut=None, mode='none'):
+        super().__init__(nlist, default_r_cut, mode)
+        params = TypeParameter(
+            'params', 'particle_types',
+            TypeParameterDict(
+                {
+                    "pair_params": self._pair_params,
+                    "envelope_params": {
+                        "alpha": OnlyTypes(float, postprocess=self._check_0_pi),
+                        "omega": float,
+                    }
+                },
+                len_keys=2))
+        envelope = TypeParameter(
+            'patches', 'particle_types',
+            TypeParameterDict([(float, float, float)], len_keys=1))
+        self._extend_typeparam((params, envelope))
+
+    @staticmethod
+    def _check_0_pi(input):
+        if 0 <= input <= np.pi:
+            return input
+        else:
+            raise ValueError(f"Value {input} is not between 0 and pi")
+        # can we get the keys here to check for A A being ni=nj
+
+
+class PatchyLJ(Patchy):
+    """Directional version of LJ."""
+    _example_doc = r"""
+    .. rubric:: Example:
+
+    .. code-block:: python
+
+        lj_params = dict(epsilon = 1, sigma = 1)
```

Comment:
```suggestion
        lj_params = dict(epsilon=1, sigma=1)
```
...and in the other code examples for consistency with the rest of the docs.